build.assets/tooling/cmd/build-apt-repos/arguments.go

```diff
@@ -0,0 +1,105 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"regexp"
+	"strings"
+)
+
+type arguments struct {
+	artifactPath    *string
+	bucketName      *string
+	localBucketPath *string
+	majorVersion    *string
+	releaseChannel  *string
+}
+
+func parseFlags() *arguments {
+	args := &arguments{
+		artifactPath:    flag.String("artifact-path", "/artifacts", "Path to the filesystem tree containing the *.deb files to add to the APT repos"),
+		bucketName:      flag.String("bucket", "", "The name of the S3 bucket where the repo should be synced to/from"),
+		localBucketPath: flag.String("local-bucket-path", "/bucket", "The local path where the bucket should be synced to"),
+		majorVersion:    flag.String("artifact-major-version", "", "The major version of the artifacts that will be added to the APT repos"),
+		releaseChannel:  flag.String("artifact-release-channel", "", "The release channel of the APT repos that the artifacts should be added to"),
+	}
+
+	flag.Parse()
+	validateArguments(args)
+	return args
+}
+
+func validateArguments(args *arguments) {
+	validateArtifactPath(args.artifactPath)
+	validateBucketName(args.bucketName)
+	validateLocalBucketPath(args.localBucketPath)
+	validateMajorVersion(args.majorVersion)
+	validateReleaseChannel(args.releaseChannel)
+}
+
+func validateArtifactPath(value *string) {
+	validateNotEmtpy(value, "artifact-path")
+
+	if stat, err := os.Stat(*value); os.IsNotExist(err) {
+		handleInvalidFlag("The artifact-path \"%s\" does not exist", *value)
+	} else if !stat.IsDir() {
+		handleInvalidFlag("The artifact-path \"%s\" is not a directory", *value)
+	}
+}
+
+func validateBucketName(value *string) {
+	validateNotEmtpy(value, "bucket")
+}
+
+func validateLocalBucketPath(value *string) {
+	validateNotEmtpy(value, "local-bucket-path")
+
+	if stat, err := os.Stat(*value); err == nil && !stat.IsDir() {
+		handleInvalidFlag("The local bucket path points to a file instead of a directory")
+	}
+}
+
+func validateMajorVersion(value *string) {
+	validateNotEmtpy(value, "artifact-major-version")
+
+	// Can somebody validate that all major versions (even for dev tags/etc.) should follow this pattern?
+	regex := `^v\d+$`
+	matched, err := regexp.MatchString(regex, *value)
```

Comment:
The main reason why I'm using regex rather than just passing in a number is that I don't know if we're going to use some other format for testing/nightly/other non-stable branches. For example, we might want a `v9-prerelease` tag. Using regex here makes it easy to check for other values should we want them.